aztec/src/main/kotlin/org/wordpress/aztec/AztecTextAccessibilityDelegate.kt

```diff
@@ -0,0 +1,107 @@
+import android.text.Selection
+import android.view.MotionEvent
+import android.view.View
+import android.view.accessibility.AccessibilityEvent
+import android.view.accessibility.AccessibilityManager
+import android.widget.EditText
+import org.apache.commons.lang3.StringUtils
+
+/**
+ * Delegate which adds support for ExploreByTouch to an EditText. TalkBack reads each line as the user hovers over them.
+ */
+class AztecTextAccessibilityDelegate(private val aztecText: EditText) {
+    private val ACCESSIBILITY_INVALID_LINE_ID = -1
+
+    private val mediaItemContentDescription = aztecText.getContext().getString(R.string.media_item_content_description)
+    private val cursorMovedText = aztecText.getContext().getString(R.string.cursor_moved)
+    private val accessibilityManager = aztecText.context.getSystemService(Context.ACCESSIBILITY_SERVICE) as AccessibilityManager
+
+    /**
+     * Offset of most recently announced line.
+     */
+    private var lastLineAnnouncedForAccessibilityOffset = ACCESSIBILITY_INVALID_LINE_ID
+
+    fun onHoverEvent(event: MotionEvent): Boolean {
+        if (!accessibilityManager.isEnabled() || !accessibilityManager.isTouchExplorationEnabled) {
+            return false
+        }
+        if (event.action == MotionEvent.ACTION_HOVER_ENTER) {
+            resetLastLineAnnouncedForAccessibilityOffset()
+        }
+        if (event.action == MotionEvent.ACTION_HOVER_EXIT) {
+            moveCursor(event.x, event.y)
+        }
+        return announceLineForAccessibility(event)
+    }
+
+    private fun resetLastLineAnnouncedForAccessibilityOffset() {
+        lastLineAnnouncedForAccessibilityOffset = ACCESSIBILITY_INVALID_LINE_ID
+    }
+
+    private fun moveCursor(x: Float, y: Float) {
+        // we need to remove the selection first, otherwise the TalkBack reads the text between old and new cursor position
+        Selection.removeSelection(aztecText.text)
+        aztecText.announceForAccessibility(cursorMovedText)
+        Selection.setSelection(aztecText.text, aztecText.getOffsetForPosition(x, y))
+    }
+
+    private fun announceLineForAccessibility(event: MotionEvent): Boolean {
+        (aztecText.parentForAccessibility as? View)?.let {
+            val lineOffset = getLineOffset(event.x, event.y)
+            if (lineOffset != ACCESSIBILITY_INVALID_LINE_ID && lastLineAnnouncedForAccessibilityOffset != lineOffset) {
+                updateContentDescription(it, lineOffset)
+            }
+            if (event.action == MotionEvent.ACTION_HOVER_EXIT) {
+                restoreContentDescription(it)
+            }
+            if (lineOffset != ACCESSIBILITY_INVALID_LINE_ID) {
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun updateContentDescription(parentForAccessibility: View, lineOffset: Int) {
+        // we can't use announceForAccessibility(..) as the announcement doesn't get interrupted as we move to another element
```

Comment:
Not sure if this will change anything, but `accessibilityManager` has `interrupt()` method :)